libraries/AP_HAL_SITL/UARTDriver.cpp

```diff
@@ -222,11 +222,16 @@ void UARTDriver::_tcp_start_connection(uint16_t port, bool wait_for_connection)
         }
         sockaddr.sin_family = AF_INET;
 
-        _listen_fd = socket(AF_INET, SOCK_STREAM | SOCK_CLOEXEC, 0);
+        _listen_fd = socket(AF_INET, SOCK_STREAM, 0);
         if (_listen_fd == -1) {
             fprintf(stderr, "socket failed - %s\n", strerror(errno));
             exit(1);
         }
+        ret = fcntl(_listen_fd, F_SETFD, FD_CLOEXEC);
+        if (ret == -1) {
+            fprintf(stderr, "socket failed on setting FD_CLOEXEC - %s\n", strerror(errno));
```

Comment:
This should be `fcntl` not `socket`.  Not a blocker.

Reply:
Done, I thought of it as setting FD_CLOEXEC failed on the socket, but it does make more sense to state which function is failing.